receiver/prometheusreceiver/metrics_receiver_helper_test.go

```diff
@@ -447,6 +447,9 @@ func compareAttributes(attributes map[string]string) numberPointComparator {
 		if req {
 			for k, v := range attributes {
 				val, ok := numberDataPoint.Attributes().Get(k)
+				if !ok {
+					log.Printf("Nothing found")
```

Comment:
I'd suggest using [t.Log](https://pkg.go.dev/testing#T.Log) for these sorts of things when in a test, that way the input is easier to follow in parallel tests and is controlled by the `-v` flag.
(doesn't matter if you use my other suggestion here)